### PR TITLE
community/rust: upgrade to 1.32.0

### PR DESCRIPTION
**Pull request by jeizsm**
@jirutka review pls

### CODE REVIEW

**Comment by jeizsm** [comment]
@ncopa @jirutka pls review

**Comment by tcely** [comment]
https://blog.rust-lang.org/2019/02/28/Rust-1.33.0.html

File: community/rust/APKBUILD [hunk 1]
```diff
@@ -3,10 +3,10 @@
 # Contributor: Jeizsm <jeizsm@gmail.com>
 # Maintainer: Jakub Jirutka <jakub@jirutka.cz>
 pkgname=rust
-pkgver=1.31.1
+pkgver=1.32.0
 # TODO: bump to 6 as soon as we add llvm6
```
Comment (J0WI) [inline community/rust/APKBUILD]: LLVM 7 has been added in #5717

Comment (jeizsm) [reply community/rust/APKBUILD]: Thx, i tried to compile 1.32 with llvm7 and lto test failed, after that I found these issue. https://github.com/rust-lang/rust/issues/57762
So it's needed to add these patches to alpine llvm7 to use it with rust
https://github.com/llvm-mirror/llvm/commit/da1fb72
https://github.com/llvm-mirror/llvm/commit/cc1f2a5

Comment (J0WI) [reply community/rust/APKBUILD]: Would this be fixed by LLVM 8?

Comment (jeizsm) [reply community/rust/APKBUILD]: Yes, but patches already in alpine llvm7 https://github.com/alpinelinux/aports/pull/6892
I'm trying to build rust with it and if everything ok, I will update this pull request with llvm7

**Comment by billiob** [comment]
Using this branch, I have the following test failures:

> failures:
    new::finds_author_email
    new::finds_author_git
    new::finds_author_user
    new::finds_author_user_escaped
    new::finds_author_username
    new::finds_git_author
    new::strip_angle_bracket_author_email
test result: FAILED. 1471 passed; 7 failed; 2 ignored; 0 measured; 0 filtered out

**Comment by andypost** [comment]
Meantime https://blog.rust-lang.org/2019/04/11/Rust-1.34.0.html

**Comment by billiob** [comment]
Once the tests are skipped, the generated packages do run fine.

**Comment by jeizsm** [comment]
@billiob Don't know why this test is failing. In my environment all tests passed

**Comment by billiob** [comment]
I compiled it again and had the same tests failing.

**Comment by mati865** [comment]
@billiob those are Cargo test failures, what is the exact error?

**Comment by billiob** [comment]
@mati865 those are cargo tests indeed:
```
failures:

---- new::finds_author_email stdout ----
running `/home/billiob/aports/community/rust/src/rustc-1.32.0-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/release/cargo new foo`
thread 'new::finds_author_email' panicked at 'assertion failed: contents.contains(r#"authors = ["bar <baz>"]"#)', src/tools/cargo/tests/testsuite/new.rs:234:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:71
             at src/libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:211
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:491
   5: std::panicking::begin_panic
   6: testsuite::new::finds_author_email
   7: <F as alloc::boxed::FnBox<A>>::call_box
             at src/libtest/lib.rs:1471
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/libcore/ops/function.rs:238
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/liballoc/boxed.rs:673
   8: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:102

---- new::finds_author_user stdout ----
running `/home/billiob/aports/community/rust/src/rustc-1.32.0-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/release/cargo new foo`
thread 'new::finds_author_user' panicked at 'assertion failed: contents.contains(r#"authors = ["foo"]"#)', src/tools/cargo/tests/testsuite/new.rs:168:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:71
             at src/libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:211
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:491
   5: std::panicking::begin_panic
   6: testsuite::new::finds_author_user
   7: <F as alloc::boxed::FnBox<A>>::call_box
             at src/libtest/lib.rs:1471
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/libcore/ops/function.rs:238
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/liballoc/boxed.rs:673
   8: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:102

---- new::finds_author_git stdout ----
running `/home/billiob/aports/community/rust/src/rustc-1.32.0-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/release/cargo new foo`
thread 'new::finds_author_git' panicked at 'assertion failed: contents.contains(r#"authors = ["bar <baz>"]"#)', src/tools/cargo/tests/testsuite/new.rs:251:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:71
             at src/libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:211
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:491
   5: std::panicking::begin_panic
   6: testsuite::new::finds_author_git
   7: <F as alloc::boxed::FnBox<A>>::call_box
             at src/libtest/lib.rs:1471
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/libcore/ops/function.rs:238
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/liballoc/boxed.rs:673
   8: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:102

---- new::finds_author_username stdout ----
running `/home/billiob/aports/community/rust/src/rustc-1.32.0-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/release/cargo new foo`
thread 'new::finds_author_username' panicked at 'assertion failed: contents.contains(r#"authors = ["foo"]"#)', src/tools/cargo/tests/testsuite/new.rs:199:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:71
             at src/libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:211
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:491
   5: std::panicking::begin_panic
   6: testsuite::new::finds_author_username
   7: <F as alloc::boxed::FnBox<A>>::call_box
             at src/libtest/lib.rs:1471
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/libcore/ops/function.rs:238
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/liballoc/boxed.rs:673
   8: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:102

---- new::finds_author_user_escaped stdout ----
running `/home/billiob/aports/community/rust/src/rustc-1.32.0-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/release/cargo new foo`
thread 'new::finds_author_user_escaped' panicked at 'assertion failed: contents.contains(r#"authors = ["foo \"bar\""]"#)', src/tools/cargo/tests/testsuite/new.rs:182:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:71
             at src/libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:211
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:491
   5: std::panicking::begin_panic
   6: testsuite::new::finds_author_user_escaped
   7: <F as alloc::boxed::FnBox<A>>::call_box
             at src/libtest/lib.rs:1471
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/libcore/ops/function.rs:238
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/liballoc/boxed.rs:673
   8: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:102

---- new::strip_angle_bracket_author_email stdout ----
running `/home/billiob/aports/community/rust/src/rustc-1.32.0-src/build/x86_64-alpine-linux-musl/stage2-tools/x86_64-alpine-linux-musl/release/cargo new foo`
thread 'new::strip_angle_bracket_author_email' panicked at 'assertion failed: contents.contains(r#"authors = ["bar <baz>"]"#)', src/tools/cargo/tests/testsuite/new.rs:354:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:71
             at src/libstd/sys_common/backtrace.rs:59
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:211
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:221
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:491
   5: std::panicking::begin_panic
   6: testsuite::new::strip_angle_bracket_author_email
   7: <F as alloc::boxed::FnBox<A>>::call_box
             at src/libtest/lib.rs:1471
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/libcore/ops/function.rs:238
             at /home/billiob/aports/community/rust/src/rustc-1.32.0-src/src/liballoc/boxed.rs:673
   8: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:102
```

**Comment by mati865** [comment]
@billiob unfortunately it's not very helpful. I think you'd have to add `println!("{:?}", contents);` [right before the assert](https://github.com/rust-lang/cargo/blob/4986d8ce8992d43908107915abe6fc89327aaee9/tests/testsuite/new.rs#L247) to peek contents of that file.

**Comment by billiob** [comment]
By running cargo tests by hand (`cargo test` from cargo's repository on branch rust-1.32), I do not have any issue.

**Comment by Ikke** [comment]
This is currently conflicting with master. CI is failing due to time, so this has to be tested locally.

@billiob Could you rebase this against the latest master? Thanks.

**Comment by billiob** [comment]
@Ikke I've tried again and the same tests do fail for me if tested within `abuild -r`.
If I don't test in the APKBUILD and test from cargo's repo with the generated package, they do pass…

Also I can't put to that branch. @jeizsm should do it.
I've pushed it to my own fork at https://github.com/billiob/aports/tree/upgrade/rust

**Comment by jeizsm** [comment]
@Ikke I rebased against the latest master
@billiob I build it in my own gitlab ci and it work just fine. [gitlab ci log](https://storage.googleapis.com/gitlab-gprd-artifacts/93/7f/937fc37e919aeeadc0f8be70eeea87d8146270388bdd5aa093383332ba07fda7/2019_04_15/196672153/202681762/job.log?response-content-type=text%2Fplain%3B%20charset%3Dutf-8&response-content-disposition=inline&GoogleAccessId=gitlab-object-storage-prd@gitlab-production.iam.gserviceaccount.com&Signature=cYx04BAIJ9hUhXSStoF0dqnAKauZ5dO%2FRPk%2Fpgp%2Fkru5oYWUoG5RFQDR%2F5lG%0AoAZW7m9HCrMHh4%2B%2FqPGGaQBd5xdg3V2M7YYdmhovgyJeKuMVJCc7FOICsRNB%0AsibZGW7mkEYRnlI2aUgondz3Il9zEC3bFugkpdbnjdFiwLZ8sgsY%2BKzFs1XY%0A%2B%2FTW%2FubM5SCINv8gjX9Wx%2FxzvZbeVZtaNOV7EjQhGTfO14yMCl%2BuLXL2Czrb%0Ax40Q8yMoBGIWqzB5%2BXh5rjbggcDn2E6vpoOLbgoR3zJuhYUQU%2FAkPeN9sKPe%0A4j0DxyzBVOK3NA7USOm%2F2Vjke%2FxB%2BacAm1anQWRUeQ%3D%3D&Expires=1558107110)

**Comment by billiob** [comment]
Now that I think again about it, I might know why it failed. On the environment where I compile the APK, I do have a `.gitconfig` file and it may mess things up.

**Comment by Cogitri** [comment]
Soo....how long are we going to wait on this? We only need Rust 1.32 to uograde to 1.33/
1.34 anyway.

**Comment by J0WI** [comment]
`rust-lldb` is broken. It depends on outdated `testing/lldb`.

**Comment by Ikke** [comment]
@Cogitri 

When I run this locally, I get some tests failures:

* new::finds_author_email                                      
* new::finds_author_git                                        
* new::finds_author_user                                                                            
* new::finds_author_user_escaped                                                              
* new::finds_author_username                       
* new::finds_git_author                      
* new::strip_angle_bracket_author_email

Which seem to panic. Do you think we should ignore that so that we can upgrade to 1.34?

**Comment by Cogitri** [comment]
How about this: I'll test if this can build Rust 1.33 and will file a PR for the Rust 1.32 -> Rust 1.33 bump based on this. Once that PR is up we can merge this and then merge the Rust 1.32 -> 1.33 PR. Then we can worry about tests once we have Rust 1.34 w/ LLVM8.

**Comment by mati865** [comment]
@Ikke it's related to your git config.

**Comment by Ikke** [comment]
@mati865 So the test is broken :-). My personal git config should not affect the outcome of any tests.

**Comment by Ikke** [comment]
I've skipped (`#[ignore]`) the broken tests. Pushed this.

**Pull request closed without merging**